libgadget/blackhole.c

```diff
@@ -766,8 +770,17 @@ blackhole_feedback_ngbiter(TreeWalkQueryBHFeedback * I,
         O->BH_CountProgs += BHP(other).CountProgs;
         /* Leave the swallowed BH mass around
          * so we can work out mass at merger. */
-        O->Mass += (P[other].Mass);
+        
         O->BH_Mass += (BHP(other).Mass);
+        
+        /* Prevent large DynMass accumulates during merger */
+        if (I->BH_Mass + BHP(other).Mass < blackhole_params.SeedBHDynMass) {
+            O->Mass += 0; 
+        }
+        else {
+            O->Mass += I->BH_Mass + BHP(other).Mass - I->Mass; 
```

Comment:
I think I am just unhappy with breaking mass conservation under any circumstances. It seems like a bad idea that will have Consequences. I think that what we really want to do is to increase the dynamical friction force for newly-seeded BHs, right? So why don't we just do that?

Reply:
Just BH born with higher dynamic mass is much more stable in the minimum potential than just boosting the dynamic friction.

Reply:
So can we modify the seeding so it uses two particles?

Reply:
Can we convert, say, 6 gas particles into BH when seeding a new blackhole? that will make mass conserves then.

Reply:
Not sure how to do this.  Implementing it in swallowing of gas particle sounds easier to me.

Reply:
That's not a bad idea: implement faster accretion just after seeding.

Reply:
We could get fancy and just implement a different seeding model. Maybe an initial mode that seeds from and accretes stars to model the runaway GC mergers that are supposed to start these things.

Reply:
ITtsounds like we need two Mass attributes to handle this. 1 The mass that is accepted by gravity, which is temporarily not conserved (set to DynSeedMass) in order to make sure the blackhole location is stable. 2. Another conserved mass that tracks the swallow and merger. Only when the swallow / merger mass >= DynSeedMass. 

On top of these too we still keep the accretion mass BH_Mass.

The accretion is already effectively faster immediately after seeding, because BH_mass < Mass. It would mean whenever a nearby gas is within the bound limit the probability of swallowing is 1, I suppose?